lib/payload/cloak.vala

```diff
@@ -99,23 +99,32 @@ namespace Frida {
 			listeners.add (readdir_listener);
 			interceptor.attach (readdir, readdir_listener);
 
-			var readdir64 = Gum.Module.find_export_by_name (libc, "readdir64");
-			if (readdir64 != null && readdir64 != readdir) {
+			/*
+			 * It seems that when using the musl CRT, the compiler incorrectly
+			 * determines that symbols with the same name, but a different
```

Comment:
Hmm, perhaps the reason is that it has `#define readdir64` to something, and that's getting picked up here?

Reply:
Oh maybe, if they had something like:
#define readdir64 readdir

Then that could throw it off. Then the compiler warning message was showing the source file before it had been pre-processed and hence showing the name collision. I have updated the comment accordingly and fixed it to wrap at 140 cols.